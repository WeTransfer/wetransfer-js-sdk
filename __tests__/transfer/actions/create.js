const routes = require('../../../src/config/routes');
const createAction = require('../../../src/transfer/actions/create');
const WTError = require('../../../src/error');

describe('Create transfer action', () => {
  let create = null;
  const mocks = {};
  beforeEach(() => {
    mocks.request = { send: jest.fn() };
    mocks.request.send.mockReturnValue({
      id: 'random-hash',
      version_identifier: null,
      state: 'uploading',
      shortened_url: 'https://we.tl/s-random-hash',
      name: 'Little kittens',
      description: null,
      size: 0,
      total_items: 0,
      items: []
    });

    create = createAction({
      routes,
      request: mocks.request
    });
  });

  it('should create a new transfer request', async () => {
    const transfer = await create({
      name: 'WeTransfer SDK'
    });
    expect(transfer).toMatchSnapshot();
  });

  it('should throw an exception if the request fails', async () => {
    mocks.request.send.mockImplementation(() =>
      Promise.reject(new Error('Network error.'))
    );

    try {
      await create({
        name: 'WeTransfer SDK'
      });
      await authorize();
    } catch (error) {
      expect(error).toBeInstanceOf(WTError);
    }
  });
});
