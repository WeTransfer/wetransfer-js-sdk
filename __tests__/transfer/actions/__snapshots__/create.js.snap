// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create transfer action should create a new transfer request 1`] = `
RemoteTransfer {
  "description": null,
  "id": "random-hash",
  "items": Array [],
  "name": "Little kittens",
  "shortened_url": "https://we.tl/s-random-hash",
  "size": 0,
}
`;
